common/src/main/java/dev/schmarrn/lighty/api/LightyHelper.java

```diff
@@ -29,21 +29,31 @@ private static boolean specialCases(Block block) {
         return block instanceof CarpetBlock;
     }
 
-    public static float getOffset(BlockState block, BlockPos pos, ClientLevel world) {
-        float offset = 0;
-        if (block.getBlock() instanceof SnowLayerBlock) { // snow layers
-            int layer = world.getBlockState(pos).getValue(SnowLayerBlock.LAYERS);
+    public static float getOffset(BlockState blockState, BlockState blockStateUp) {
+        // Returns the offset of blocks that aren't 16 pixels high, for example snow layers or farmland.
+        Block blockUp = blockStateUp.getBlock();
+        if (blockUp instanceof SnowLayerBlock) { // snow layers
+            int layer = blockStateUp.getValue(SnowLayerBlock.LAYERS);
             if (layer != 1) {
                 // Mobs cannot spawn on those blocks
                 return -1;
             }
             // One layer of snow is two pixels high, with one pixel being 1/16
-            offset = 2f / 16f * layer;
+            return 2f / 16f * layer;
         }
-        return offset;
+
+        Block block = blockState.getBlock();
+        if (block instanceof FarmBlock) { // farmland
+            // FarmBlocks are 15 pixels high
+            return -1f/16f;
+        }
+
+        return 0f;
     }
 
     public static boolean isBlocked(BlockState block, BlockState up, ClientLevel world, BlockPos pos, BlockPos upPos) {
+        if (block.getBlock() instanceof FarmBlock) return false; // farmland
```

Comment:
we don't need this with the dedicated farmland mode anymore, and I'll work on the farmland mode being a mode that can be on even when other modes are on